src/sql/workbench/api/browser/mainThreadDataProtocol.ts

```diff
@@ -525,7 +525,7 @@ export class MainThreadDataProtocol extends Disposable implements MainThreadData
 			generateScript(tableInfo: azdata.designers.TableInfo): Thenable<string> {
 				return self._proxy.$generateScriptForTableDesigner(handle, tableInfo);
 			},
-			generatePreviewReport(tableInfo: azdata.designers.TableInfo): Thenable<string> {
+			generatePreviewReport(tableInfo: azdata.designers.TableInfo): Thenable<string | { report: string, format: string }> {
```

Comment:
I'd suggest making a defined type for this so you don't have to copy it over in a bunch of places